packages/get-starknet/src/wallet.ts

```diff
@@ -220,13 +236,108 @@ export class MetaMaskSnapWallet implements StarknetWindowObject {
+      this.#startPolling();
+    }
+
+    if (event === 'accountsChanged') {
+      this.#accountChangeHandlers.push(handleEvent as AccountChangeEventHandler);
+    } else if (event === 'networkChanged') {
+      this.#networkChangeHandlers.push(handleEvent as NetworkChangeEventHandler);
+    } else {
+      throw new Error(`Unsupported event: ${String(event)}`);
+    }
+  }
+
+  /**
+   * Removes an event handler for `accountsChanged` or `networkChanged`.
+   *
+   * @param event - The event name ('accountsChanged' or 'networkChanged')
+   * @param _handleEvent
+   */
+  off<Event extends keyof WalletEventHandlers>(event: Event, _handleEvent?: WalletEventHandlers[Event]): void {
+    if (event === 'accountsChanged') {
+      this.#accountChangeHandlers = [];
+    } else if (event === 'networkChanged') {
+      this.#networkChangeHandlers = [];
+    } else {
+      throw new Error(`Unsupported event: ${String(event)}`);
+    }
+
+    if (this.#accountChangeHandlers.length === 0 && this.#networkChangeHandlers.length === 0) {
+      this.#stopPolling();
+    }
+  }
+
+  /**
+   * Polling function for detecting changes with a maximum delay.
+   * Ensures that each iteration respects a maximum delay before proceeding.
+   */
+  #pollingFunction = async (): Promise<void> => {
+    if (!this.#pollingController) {
+      return; // Abort if the polling controller is not initialized
+    }
+    const { signal } = this.#pollingController;
+
+    while (!signal.aborted) {
+      const startTime = Date.now(); // Track the start time of the iteration
+      const previousNetwork = this.#network;
+      const previousAddress = this.#latestAddress;
+
+      try {
+        // Perform the polling operation with a timeout
+        await Promise.race([
+          this.#init(), // Perform the main operation
+          new Promise(
+            (_, reject) =>
+              setTimeout(() => reject(new Error('Polling timeout exceeded')), MetaMaskSnapWallet.maximumPollingDelay), // Timeout after 5 seconds
+          ),
+        ]);
+
+        // Check for network change
+        if (previousNetwork.chainId !== this.#network.chainId) {
+          this.#networkChangeHandlers.forEach((callback) => callback(this.#network.chainId, [this.#selectedAddress]));
+        }
+
+        // Check for account change
+        if (previousAddress !== this.#selectedAddress) {
+          this.#accountChangeHandlers.forEach((callback) => callback([this.#selectedAddress]));
+        }
+      } catch (error) {
+        if (!signal.aborted) {
+          console.error('Error in polling:', error.message);
+        }
+      }
+
+      // Calculate remaining time for minimum delay
+      const elapsedTime = Date.now() - startTime;
+      const remainingTime = Math.max(0, MetaMaskSnapWallet.minimumPollingDelay - elapsedTime); // Ensure a minimum delay of 100ms
+      if (remainingTime > 0) {
+        await new Promise((resolve) => setTimeout(resolve, remainingTime));
+      }
```

Comment:
always have a small buffer in between each loop, wdyt

```suggestion
     await new Promise((resolve) => setTimeout(resolve, min(MetaMaskSnapWallet.minimumPollingDelay, remainingTime)));

```